### ISSUE TITLE
fix(rules): enforce descendantOf constraint in permitted-contents rule

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3670]
## Describe the bug

The `descendantOf` field in `ContentModel` is defined in the type system and used in spec data (`spec.area.jsonc`: `"descendantOf": "map"`), but the `permitted-contents` rule never checks it. The logic to enforce this constraint is missing.

## Code Example

```html
<area alt="x" href="#">
```

## The current behavior

No violations reported.

## The expected behavior

A violation reporting that the `<area>` element must appear as a descendant of a `<map>` element.

## Affected elements

Currently only `<area>` uses `descendantOf: "map"`.
